homeassistant/components/velbus/light.py

```diff
@@ -0,0 +1,85 @@
+"""Support for Velbus light dimmers."""
+import logging
+
+from velbus.util import VelbusException
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_TRANSITION,
+    SUPPORT_BRIGHTNESS,
+    SUPPORT_TRANSITION,
+    Light,
+)
+
+from . import VelbusEntity
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Old way."""
+    pass
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Velbus light based on config_entry."""
+    cntrl = hass.data[DOMAIN][entry.entry_id]["cntrl"]
+    modules_data = hass.data[DOMAIN][entry.entry_id]["light"]
+    entities = []
+    for address, channel in modules_data:
+        module = cntrl.get_module(address)
+        entities.append(VelbusLight(module, channel))
+    async_add_entities(entities)
+
+
+class VelbusLight(VelbusEntity, Light):
+    """Representation of a light dimmer."""
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_BRIGHTNESS | SUPPORT_TRANSITION
+
+    @property
+    def is_on(self):
+        """Return true if the dimmer is on."""
+        return self._module.is_on(self._channel)
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._module.get_dimmer_state(self._channel)
+
+    def turn_on(self, **kwargs):
+        """Instruct the Velbus light to turn on."""
+        if ATTR_TRANSITION in kwargs:
+            transitiontime = int(kwargs[ATTR_TRANSITION])
+        else:
+            transitiontime = 0
+        try:
+            if ATTR_BRIGHTNESS in kwargs:
+                self._module.set_dimmer_state(
+                    self._channel, int(kwargs[ATTR_BRIGHTNESS]), transitiontime,
+                )
+            else:
+                self._module.restore_dimmer_state(self._channel, transitiontime)
+        except VelbusException as err:
+            _LOGGER.error("A Velbus error occurred: %s", err)
+
+    def turn_off(self, **kwargs):
+        """Instruct the velbus light to turn off."""
+        if ATTR_BRIGHTNESS in kwargs:
```

Comment:
Brightness is not valid for turn off.

Reply:
Thanks I didn't know that, but it sounds very logical.
Fixed in [this ](https://github.com/home-assistant/home-assistant/pull/30184/commits/711d9ef5dc88f6b8310445004b7174547e544ce2) additional commit